docs/contributing/style-guide.md

```diff
@@ -1,7 +1,405 @@
+
+* H1 headers should never be used in a document since the title is automatically generated as an H1.
+* H2 headers are used for SEO, so make sure they succinctly represent what a user will find on the page in that section
+* H3 headers are included in the page's table of contents on the right, so make sure the title describes something
+a user might want to click on
+* H4 headers are to emphasize things within a subsection of the page; these can be longer than the other headers if needed. 
+
+### Code Example
+
+```markdown
+## H2 Header
+### H3 Header
+#### H4 Header
+```
+
+## Content
+
+All words are rendered in the same paragraph even with line breaks, so long as there isn't an empty line.
+As such, it is good practice for each line to be less than 120 characters long for readability, when possible.
+
+### Code Example
+
+```markdown
+This
+will
+all
+be
+on
+one
+line.
+
+The empty line above creates a new paragraph.
+
+This forces a soft return<br />
+rather than creating a new paragraph
+```
+
+### Display
+
+This
+will
+all
+be
+on
+one
+line.
+
+The empty line above creates a new paragraph.
+
+This forces a soft return<br />
+rather than creating a new paragraph
+
+## Images
+
+All image files should be included in the `sauce-docs/static/img` directory, in a sub-directory that 
+corresponds to the referencing directory. (e.g., images for a document in the `sauce-docs/docs/contributing` directory 
+would be located in the `sauce-docs/static/img/contributing` directory.
+
+To add an image from that directory, you need to import a special method by placing this line
+below the [Frontmatter](#frontmatter), but above the [Introduction](#introduction)
+
+```markdown
+import useBaseUrl from '@docusaurus/useBaseUrl';
+```
+
+and then reference the image as follows:
+```markdown
+<img src={useBaseUrl('img/contributing/my-image.png')} alt="All images should have alt text" width="250"/>
+```
+
+## Videos
+
+Any referenced videos need to be from a Sauce Labs YouTube account.
+The suggested iframe code structure is as follows:
+```markdown
+<iframe width="560" height="315" src="https://www.youtube.com/embed/-RDh1ukLN8w" frameborder="0" allow="accelerometer; 
+autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+```
+
+## Tabs
+
+Tabs are a great option for when an example is different in different contexts. The primary usage of
+Tabs on this site is for code-specific descriptions or examples, since many such things on our platform
+are relevant to multiple languages.
+Ideally these examples will exist in Java, Node.js, Python, Ruby, C#, and the tabs should be placed in that order.
+
+To use tabs, you need to import two special methods by placing these lines below the [Frontmatter](#frontmatter),
+but above the [Introduction](#introduction):
+```markdown
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+### Code Example:
+```markdown
+<Tabs
+  defaultValue="java"
```

Comment:
```suggestion
<Tabs
  groupId="lang-ex"
  defaultValue="java"
```

Reply:
So... this is really cool and I didn't realize it, need to go update my other projects with it. Great tip.